src/main/scala/chisel3/util/circt/PlusArgsValue.scala

```diff
@@ -21,7 +22,7 @@ object PlusArgsValue {
     * b.value
     * }}}
     */
-  def apply[T <: Data](gen: T, str: String) = {
+  def apply[T <: Data](gen: T, str: String)(implicit sourceInfo: SourceInfo) = {
```

Comment:
I hate to have to point out (because the fix is a little annoying), but this is a source incompatible change. If a user were doing `PlusArgsValue(UInt(8.W), "whatever")(7)` to take just the 7th bit of the value, the code will no longer compile. This is the whole `do_apply` pattern. Naturally this is fixed in Scala 3, at the moment we have to do the macro.

Reply:
I'll take a look at that, thanks, that totally makes sense.